packages/core/src/utils/checkpoint.test.ts

```diff
@@ -23,112 +83,155 @@ test("isCheckpointEqual returns false if checkpoints are different", () => {
     blockTimestamp: 1,
     chainId: 1,
     blockNumber: 1,
-    logIndex: 1,
+    eventType: 1,
+    transactionIndex: 1,
+    eventIndex: 1,
   };
   const isEqual = isCheckpointEqual(checkpoint, { ...checkpoint, chainId: 2 });
 
   expect(isEqual).toBe(false);
 });
 
-test("isCheckpointEqual returns true with logIndex undefined", () => {
-  const checkpoint = {
+test("isCheckpointGreaterThan compares correctly on blockTimestamp", () => {
+  const checkpointOne = {
     blockTimestamp: 1,
     chainId: 1,
     blockNumber: 1,
+    eventType: 1,
+    transactionIndex: 1,
+    eventIndex: 1,
+  };
+  const checkpointTwo = {
+    blockTimestamp: 2,
+    chainId: 2,
+    blockNumber: 1,
+    eventType: 1,
+    transactionIndex: 1,
+    eventIndex: 1,
   };
-  const isEqual = isCheckpointEqual(checkpoint, checkpoint);
+  const isGreater = isCheckpointGreaterThan(checkpointOne, checkpointTwo);
 
-  expect(isEqual).toBe(true);
+  expect(isGreater).toBe(false);
 });
 
-test("isCheckpointEqual returns treats undefined logIndex as distinct", () => {
-  const checkpoint = {
+test("isCheckpointGreaterThan compares correctly on chainId", () => {
+  const checkpointOne = {
     blockTimestamp: 1,
     chainId: 1,
     blockNumber: 1,
-    logIndex: 1,
+    eventType: 1,
+    transactionIndex: 1,
+    eventIndex: 1,
   };
-  const isEqual = isCheckpointEqual(checkpoint, {
-    ...checkpoint,
-    logIndex: undefined,
-  });
+  const checkpointTwo = {
+    blockTimestamp: 1,
+    chainId: 2,
+    blockNumber: 1,
+    eventType: 1,
+    transactionIndex: 1,
+    eventIndex: 1,
+  };
+  const isGreater = isCheckpointGreaterThan(checkpointOne, checkpointTwo);
 
-  expect(isEqual).toBe(false);
+  expect(isGreater).toBe(false);
 });
 
-test("isCheckpointGreaterThan compares correctly on chainId", () => {
+test("isCheckpointGreaterThan compares correctly on transactionIndex", () => {
   const checkpointOne = {
     blockTimestamp: 1,
     chainId: 1,
     blockNumber: 1,
-    logIndex: 1,
+    eventType: 1,
+    transactionIndex: 5,
+    eventIndex: 1,
```

Comment:
Super nit, but I'd probably order the properties of this object the same as the encoding order